src/storage/src/source/metrics.rs

```diff
@@ -411,11 +426,13 @@ impl UpsertMetrics {
 #[derive(Debug)]
 pub(crate) struct UpsertSharedMetrics {
     pub(crate) merge_snapshot_latency: DeleteOnDropHistogram<'static, Vec<String>>,
-    pub(crate) merge_snapshot_updates: DeleteOnDropHistogram<'static, Vec<String>>,
+    pub(crate) merge_snapshot_updates: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
+    pub(crate) merge_snapshot_inserts: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
+    pub(crate) merge_snapshot_deletes: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
     pub(crate) multi_get_latency: DeleteOnDropHistogram<'static, Vec<String>>,
-    pub(crate) multi_get_size: DeleteOnDropHistogram<'static, Vec<String>>,
+    pub(crate) multi_get_size: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
     pub(crate) multi_put_latency: DeleteOnDropHistogram<'static, Vec<String>>,
-    pub(crate) multi_put_size: DeleteOnDropHistogram<'static, Vec<String>>,
+    pub(crate) multi_put_size: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
```

Comment:
the counters should be moved out of the `shared` metrics and have `worker_id` added to them!